### ISSUE TITLE
Error while terminating (CUDA DEBUG compilation + Particles)

### ISSUE THREAD

**Issue by debog** [erf-model/ERF#1419]
When a simulation with particles (for example, `particlesWoA`) is run on GPUs with the code compiled with `DEBUG=TRUE`, it shows the following error at exit:
```
AMReX (24.01-30-g41397e2e7049) finalized
pure virtual method called
terminate called without an active exception
[lassen33:126238] *** Process received signal ***
[lassen33:126238] Signal: Aborted (6)
[lassen33:126238] Signal code:  (-33568)

```
I haven't been able to understand why this is happening. This doesn't happen with 
+ non-GPU code (compiled with `DEBUG=TRUE` or not)
+ on GPU but not compiled with `DEBUG=TRUE`.

**Comment by gardner48** [comment]
@debog looks like @asalmgren has a fix for this in PR #1422

**Comment by asalmgren** [comment]
yes -- I've tested that this fix addresses the bug you pointed out -- thanks for catching that!  Fix is entirely courtesy of Weiqun :-)
